### ISSUE TITLE
Model not learning at all...

### ISSUE THREAD

**Issue by pgaston1** [ybkscht/EfficientPose#33]
Similar to others, when I train the model it returns values of 0.0000 and nan.     I did review issue #23 -- I did validate that both 2D and 3D bounding boxes are shown using debug.py.     However, there was a magic step there that I missed to get to a solution (I believe I am using a COCO weights file.)    (I even removed all and then retried again following instructions.)   I'm on Ubuntu 18, GPU is RTX 3070, installed Cuda 10.2.     

Attached are a file showing a complete run, and the 'conda list' results.  [EffPose.txt](https://github.com/ybkscht/EfficientPose/files/7062104/EffPose.txt)
[EPCondaList.txt](https://github.com/ybkscht/EfficientPose/files/7062105/EPCondaList.txt)

I launched training with: (I've tried w/ no weights specified, and other variants - no change...)
`python train.py --phi 0 --weights /media/pg/2TB/data/EfficientPose/Weights/COCO/efficientdet-d0.h5 linemod /media/pg/2TB/data/EfficientPose/Linemod_preprocessed/ --object-id 8
`

Here is the core problem:
```
1009 instances of class object with average precision: 0.0000
[EffPose.txt](https://github.com/ybkscht/EfficientPose/files/7062072/EffPose.txt)

1009 instances of class object with ADD accuracy: 0.0000
1009 instances of class object with ADD-S-Accuracy: 0.0000
1009 instances of class object with 5cm-5degree-Accuracy: 0.0000
[EPCondaList.txt](https://github.com/ybkscht/EfficientPose/files/7062073/EPCondaList.txt)

class object with Translation Differences in mm: Mean: nan and Std: nan
class object with Rotation Differences in degree: Mean: nan and Std: nan
1009 instances of class object with 2d-projection-Accuracy: 0.0000
1009 instances of class object with ADD(-S)-Accuracy: 0.0000
class object with Transformed Point Distances in mm: Mean: nan and Std: nan
class object with Transformed Symmetric Point Distances in mm: Mean: nan and Std: nan
class object with Mixed Transformed Point Distances in mm: Mean: nan and Std: nan
WARNING:tensorflow:From /home/pg/repos/EfficientPose/eval/eval_callback.py:240: The name tf.Summary is deprecated. Please use tf.compat.v1.Summary instead.

mAP: 0.0000
ADD: 0.0000
ADD-S: 0.0000
5cm_5degree: 0.0000
TranslationErrorMean_in_mm: nan
TranslationErrorStd_in_mm: nan
RotationErrorMean_in_degree: nan
RotationErrorStd_in_degree: nan
2D-Projection: 0.0000
Summed_Translation_Rotation_Error: nan
ADD(-S): 0.0000
AveragePointDistanceMean_in_mm: nan
AveragePointDistanceStd_in_mm: nan
AverageSymmetricPointDistanceMean_in_mm: nan
AverageSymmetricPointDistanceStd_in_mm: nan
MixedAveragePointDistanceMean_in_mm: nan
MixedAveragePointDistanceStd_in_mm: nan

Epoch 00001: ADD improved from -inf to 0.00000, saving model to checkpoints/26_08_2021_13_58_29/object_8/phi_0_linemod_best_ADD.h5
1790/1790 [==============================] - 1561s 872ms/step - loss: nan - classification_loss: 16538.6211 - regression_loss: nan - transformation_loss: 0.0000e+00
Epoch 2/500

```
And that continues, only classification_loss changes - and rather than be < 1.0 it's in the ten thousands.

Thanks!

**Comment by pgaston1** [comment]
Update - trying inferency.py (with folders appropriately set)

- fails w/ GPU - and takes roughly 10 minutes at one point
- works great w/ CPU.

I'm now on the track that the RTX3070 is not supported by cuda 10.0 - which seems to have been called for by Tensorflow 1.15.

Ughh. 

(btw, using opencv 4.3.0.38 (rather than 36) fixes an imshow bug.)

**Comment by pgaston1** [comment]
Okay, I rebuilt another machine with a lesser 'compute capability' - not in the Ampere class - and everything works as planned.    i.e., I'm claiming it's the Ampere class machine forcing higher versions of CUDA...

**Comment by enesdoruk** [comment]
use coco pretrained weights, it will be solvec

**Comment by takuma-yoneda** [comment]
Even using coco pretrained weights didn't work for me using `tensorflow/tensorflow:1.15.0-gpu-py3` docker image.
But throwing out my A4000 (Ampere) and using the old RTX8000 (Turing) got it to work 🙃.
The loss is not nan anymore.
